src/experiments.js

```diff
@@ -54,6 +58,88 @@ export function isCanary(win) {
   return !!(win.AMP_CONFIG && win.AMP_CONFIG.canary);
 }
 
+/**
+ * Whether the specified experiment is on or off for origin trials. You should
+ * check if the experiment is already enabled before calling this function.
+ * @param {!Window} win
+ * @param {string} experimentId
+ * @param {!Object} opt_publicJwk Used for testing only.
+ * @return {!Promise<boolean>}
+ */
+export function isExperimentOnForOriginTrial(win, experimentId, opt_publicJwk) {
+  const crypto = cryptoFor(win);
+  const meta =
+      win.document.head.querySelector('meta[name="amp-experiment-token"]');
+  if (!meta || !crypto.isCryptoAvailable()) {
+    return Promise.resolve(false);
+  }
+  const token = meta.getAttribute('content');
+  if (!token) {
+    return Promise.reject(new Error('Unable to read experiments token'));
+  }
+  /**
+   * token = encode64(version + length + config + sign(config, private_key))
+   * version = 1 byte version of the token format (starting at 0x0)
+   */
+  const bytes = stringToBytes(atob(token));
+  const version = bytes[0];
+  if (version === 0) {
+   /**
+    * Version 0:
+    * length = 4 bytes representing number of bytes in config
+    * config = string containing the experiment ID, origin URL, etc.
+    */
+    const configLen = bytesToInt(bytes.subarray(1,5));
+    if (configLen > bytes.length - 5) {
+      return Promise.reject(
+          new Error('Specified len extends past end of buffer'));
+    }
+    const configBytes = bytes.subarray(5, 5 + configLen);
+    const signatureBytes = bytes.subarray(5 + configLen);
+
+    // TODO(kmh287, choumx) fill in real public key
+    const publicJwk = opt_publicJwk || {};
+
+    return crypto.importPublicKey('experiments', publicJwk).then(keyInfo => {
+      return crypto.verifySignature(configBytes, signatureBytes, keyInfo);
+    }).then(verified => {
+      if (!verified) {
+        throw new Error('Failed to verify config signature');
+      }
+      const configStr = utf8DecodeSync(configBytes);
+      const config = JSON.parse(configStr);
+
+      const approvedOrigin = config['origin'];
+      const url = ampdocServiceFor(win).getAmpDoc().getUrl();
+      const sourceOrigin = getSourceOrigin(url);
+      if (approvedOrigin !== sourceOrigin) {
+        throw new Error('Config does not match current origin');;
+      }
+
+      const experiments = config['experiments'];
+      const experiment = experiments[experimentId];
+      if (!experiment) {
+        return false;
+      }
+
+      const expiration = experiment['expiration'];
+      const now = Date.now();
+      if (expiration < now) {
+        return false;
```

Comment:
I think there are three different cases here:

1. The token is correct and the experiment is enabled so we return true
2. The token is correct and the experiment is expired or not enabled, so we return false
3. The token is malformed, so we return an error.

Consider a page that has an experiments token but does not have `amp-bind` enabled. Having this method reject almost feels like using exceptions for control flow. WDYT?